### PR TITLE
Lesson 07

### PR DESCRIPTION
**Pull request by StrafingRun**
Lesson 07 ready for review.

### CODE REVIEW

**Comment by StrafingRun** [comment]
Lesson 08 is ready for review as well.

File: students/grant_dowell/Lesson_07/html_render.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    tag = "html"
+    tag_attributes = dict()
```
Comment (natasha-aleksandrova) [inline students/grant_dowell/Lesson_07/html_render.py]: this should not be a class variable --- if you are setting it via `__init__` method then it's an instance attribute.

File: students/grant_dowell/Lesson_08/circle.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        
+    @property
+    def radius(self):
+        return self._radius
+    @radius.setter
+    def radius(self, new_radius):
+        if new_radius >= 0:
+            self._radius = new_radius
+        else:
+            raise ValueError
```
Comment (natasha-aleksandrova) [inline students/grant_dowell/Lesson_08/circle.py]: this is fine and works, here's a more common pattern:
```
if new_radius < 0:
    raise ValueError

self._radius = new_radius
```

**Comment by natasha-aleksandrova** [comment]
make sure to submit on the course website as well

**Pull request merged**